src/lib/rest/curl_context.cpp

```diff
@@ -0,0 +1,94 @@
+#include "curl_context.h"
+
+#include <map>
+
+#include <assert.h>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+
+static pthread_mutex_t contexts_mutex = PTHREAD_MUTEX_INITIALIZER;
+static std::map<std::string, struct curl_context> contexts;
+
+struct curl_context
```

Comment:
Yes, the return type and function name we keep on the same line, like this:

    XXX* xxx(void)
    {
      ...
    }

Some functions have A LOT of parameters and we then put each parameter on a separate line:

    int xxx
    (
        int i1,
        int i2
    )
    {
      …
    }

Two spaces used as indentation. Tabs are not used. Ever. etc ...